components/CollectionList/CollectionList.js

```diff
@@ -14,25 +14,31 @@ export default function CollectionList({
   return (
     <>
       {collections &&
-        collections.map((collection) => {
-          const filteredCards = cards.filter(
-            (card) => card.collection === collection.id
-          );
-          return (
-            <StyledListContainer key={collection.id}>
-              <CollectionContainer
-                collection={collection}
-                editCollection={editCollection}
-                deleteCollection={deleteCollection}
-                cards={filteredCards}
-                archive={archive}
-                editCard={editCard}
-                resetCard={resetCard}
-                toggleCardDirection={toggleCardDirection}
-              />
-            </StyledListContainer>
-          );
-        })}
+        collections
```

Comment:
Consider moving complex logic outside the return statement to keep the JSX clean. 

_Here is a suggestion on how you could implement it_

```js


// 📦 store the sorted collection in a variable 
// ⚠️ preserving the immutability of the collections array by using the spread operator
  const sortedCollections = collections ? [...collections].sort((c1, c2) => {
    if (c1.timestamp > c2.timestamp) return -1;
    if (c1.timestamp < c2.timestamp) return 1;
    return 0;
  }) : [];

// 📦 store the sorted collection items in a variable. 
//  iterate over sortedCollections 
const collectionItems = sortedCollections.map(collection => {
    const filteredCards = cards.filter(card => card.collection === collection.id);

```

Reply:
@JessicaLoers  Dear Jessica, thank you for your tip! I've learned something about best practice again! 😇 I've implemented it now!😊